addons/base_iban/static/src/tests/iban_widget.test.js

```diff
@@ -0,0 +1,77 @@
+import { DELAY } from "@base_iban/components/iban_widget/iban_widget";
+import { test } from "@odoo/hoot";
+import { advanceTime } from "@odoo/hoot-mock";
+import {
+    click,
+    contains,
+    defineMailModels,
+    insertText,
+    openFormView,
+    start,
+    startServer,
+} from "@mail/../tests/mail_test_helpers";
+import { defineModels, models, onRpc } from "@web/../tests/web_test_helpers";
+
+class ResPartnerBank extends models.ServerModel {
+    _name = "res.partner.bank";
+}
+
+defineModels([ResPartnerBank]);
+defineMailModels();
+
+const [validIban, invalidIban] = ["BE12651194580992", "invalidIban!"];
+
+test("Iban Widget full flow [REQUIRE FOCUS]", async () => {
+    const pyEnv = await startServer();
+    const bankId = pyEnv["res.partner.bank"].create({ acc_number: "" });
+    const partnerId = pyEnv["res.partner"].create({
+        name: "Awesome partner",
+        bank_ids: [bankId],
+    });
+    await start();
+    onRpc(async (params) => {
```

Comment:
async does not look required here? also if you want to listen to a single method, better use the `onRpc` ORM syntax (`onRpc("check_iban")` and to get rid of the if below. Finally, I don't think that `Promise.resolve` is required, you can just return the value.